util/file.py

```diff
@@ -339,3 +339,51 @@ def temp_catted_files(input_files, prefix=None, suffix=None):
         yield fn
     finally:
         os.remove(fn)
+
+def string_to_file_name(string_value):
+    replacements_dict = {
+        "\\": "-", # win directory separator 
+        "/": "-", # posix directory separator 
+        "^": "_", # caret
+        "&": "_and_", # background
+        "\"": "", # double quotes
+        r"'": "", # single quotes
+        r":": "_", # colon (problem for ntfs)
+        r" ": "_", # spaces
+        r"|": "-", # shouldn't confuse a vertical bar for a shell pipe
+        r"!": ".", # not a bash operator
+        r";": ".", # not a terminator
+        r"?": "_", # could be mistaken for a wildcard
+        r"*": "_", # could be mistaken for a wildcard
+        r"`": "_", # no subshells
+        r" -": "_-", # could be mistaken for an argument
+        r" --": "_--", # could be mistaken for an argument
+        r">": "]", # no redirect chars
+        r"<": "[", # no redirect chars
+        r"\\x": "_", # hex char
+        r"\\o": "_", # octal char
+        r"\\u": "", # unicode char
```

Comment:
Just encoded unicode. We could omit line 365; I'll comment it out for now. Under py3 unicode passes through the function unchanged, but now that I've added "Sénégal" and some other unicode to the test case I see there are failures elsewhere. I took a look at the bam file created by the unit test, and seems looks like samtools writes latin-1 but strips utf-8? I've specified the encoding in samtools.py getHeader() as "latin-1" and we can at least load "Sénégal" now. We should, in general, talk about our encoding expectations for viral-ngs...